winpr/libwinpr/utils/collections/Reference.c

```diff
@@ -78,10 +81,12 @@ wReference* ReferenceTable_GetFreeEntry(wReferenceTable* referenceTable)
 			return NULL;
 		}
 
-		referenceTable->size *= 2;
-		referenceTable->array = (wReference*) realloc(referenceTable->array,
-				sizeof(wReference) * referenceTable->size);
+		new_size = referenceTable->size * 2;
+		new_ref = (wReference*) realloc(referenceTable->array,
+				sizeof(wReference) * new_size);
 
```

Comment:
new_ref is not check and that would segfault two lines later when ZeroMemorying referenceTable->array